src/test/java/seedu/duke/command/CommandAddClientTest.java

```diff
@@ -0,0 +1,89 @@
+
+import seedu.duke.ClientList;
+import seedu.duke.PairingList;
+import seedu.duke.PropertyList;
+import seedu.duke.Storage;
+import seedu.duke.Ui;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CommandAddClientTest {
+    private static final String SPACE = " ";
+    private static final String DOUBLE_SPACE = SPACE + SPACE;
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String CLIENT_NAME_LABEL = "Client:";
+    private static final String CLIENT_CONTACT_NUMBER_LABEL = "Contact Number:";
+    private static final String CLIENT_EMAIL_LABEL = "Email:";
+    private static final String CLIENT_BUDGET_LABEL = "Budget: SGD";
+    private static final String CLIENT_PER_MONTH_LABEL = "/month";
+
+    //Test Case 1 (Client with Email)
+    private static final String CLIENT_ONE_NAME = "Joseph Joestar";
+    private static final String CLIENT_ONE_CONTACT_NUMBER = "83625471";
+    private static final String CLIENT_ONE_EMAIL = "jojofirst@jmail.com";
+    private static final String CLIENT_ONE_BUDGET = "9000";
+    private static final ArrayList<String> CLIENT_ONE_DETAILS = new ArrayList<>(
+            List.of(CLIENT_ONE_NAME, CLIENT_ONE_CONTACT_NUMBER, CLIENT_ONE_EMAIL, CLIENT_ONE_BUDGET)
+    );
+    private static final String CLIENT_ONE_SUMMARY = CLIENT_NAME_LABEL + SPACE + CLIENT_ONE_NAME
+            + LINE_SEPARATOR + DOUBLE_SPACE + CLIENT_CONTACT_NUMBER_LABEL + SPACE + CLIENT_ONE_CONTACT_NUMBER
+            + LINE_SEPARATOR + DOUBLE_SPACE + CLIENT_EMAIL_LABEL + SPACE + CLIENT_ONE_EMAIL
+            + LINE_SEPARATOR + DOUBLE_SPACE + CLIENT_BUDGET_LABEL + CLIENT_ONE_BUDGET + CLIENT_PER_MONTH_LABEL;
+
+    //Test Case 2 (Client without Email)
+    private static final String CLIENT_TWO_NAME = "Kujou Jotaro";
+    private static final String CLIENT_TWO_CONTACT_NUMBER = "65329466";
+    private static final String CLIENT_TWO_EMAIL = "";
+    private static final String CLIENT_TWO_BUDGET = "10000";
+    private static final ArrayList<String> CLIENT_TWO_DETAILS = new ArrayList<>(
+            List.of(CLIENT_TWO_NAME, CLIENT_TWO_CONTACT_NUMBER, CLIENT_TWO_EMAIL, CLIENT_TWO_BUDGET)
+    );
+    private static final String CLIENT_TWO_SUMMARY = CLIENT_NAME_LABEL + SPACE + CLIENT_TWO_NAME
+            + LINE_SEPARATOR + DOUBLE_SPACE + CLIENT_CONTACT_NUMBER_LABEL + SPACE + CLIENT_TWO_CONTACT_NUMBER
+            + LINE_SEPARATOR + DOUBLE_SPACE + CLIENT_BUDGET_LABEL + CLIENT_TWO_BUDGET + CLIENT_PER_MONTH_LABEL;
+
+    //Initialization for Testing
+    public Ui ui = new Ui();
+    public PropertyList propertyList = new PropertyList();
+    public ClientList clientList = new ClientList();
+    public PairingList pairingList = new PairingList();
+    public Storage storage = new Storage(clientList, propertyList, pairingList);
+
+    @Test
+    public void execute() {
```

Comment:
Do follow the convention of whatIsBeingTested_descriptionOfTestInputs_expectedOutcome for test methods